From: David Teigland <teigland@redhat.com>
commit 24179f488092267c9a033d7e25ce7a58af50ff79
Author: David Teigland <teigland@redhat.com>
Date:   Mon Jan 19 13:13:33 2009 -0600
Subject: dlm: fix plock notify callback to lockd
    
    We should use the original copy of the file_lock, fl, instead
    of the copy, flc in the lockd notify callback.  The range in flc has
    been modified by posix_lock_file(), so it will not match a copy of the
    lock in lockd.
    
Signed-off-by: David Teigland <teigland@redhat.com>
Signed-off-by: Coly Li <coly.li@suse.de>

diff --git a/fs/dlm/plock.c b/fs/dlm/plock.c
index eba87ff..502b1ea 100644
--- a/fs/dlm/plock.c
+++ b/fs/dlm/plock.c
@@ -168,7 +168,7 @@ static int dlm_plock_callback(struct plock_op *op)
 	notify = xop->callback;
 
 	if (op->info.rv) {
-		notify(flc, NULL, op->info.rv);
+		notify(fl, NULL, op->info.rv);
 		goto out;
 	}
 
@@ -187,7 +187,7 @@ static int dlm_plock_callback(struct plock_op *op)
 			  (unsigned long long)op->info.number, file, fl);
 	}
 
-	rv = notify(flc, NULL, 0);
+	rv = notify(fl, NULL, 0);
 	if (rv) {
 		/* XXX: We need to cancel the fs lock here: */
 		log_print("dlm_plock_callback: lock granted after lock request "
